woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-tokens-handler.php

```diff
@@ -613,8 +613,28 @@ public function update_tokens( $user_id, $tokens, $environment_id = null ) {
 		// clear the transient
 		$this->clear_transient( $user_id );
 
-		// persist the updated tokens to the user meta
-		return update_user_meta( $user_id, $this->get_user_meta_name( $environment_id ), $this->format_for_db( $tokens ) );
+		$updated_tokens = [];
+
+		foreach ( $tokens as $token ) {
+
+			// skip invalid objects
+			if ( ! $token instanceof SV_WC_Payment_Gateway_Payment_Token ) {
+				continue;
+			}
+
+			// ensure the vital properties are set
+			$token->set_user_id( $user_id );
+			$token->set_gateway_id( $this->get_gateway()->get_id() );
+			// TODO: set the environment ID {CW 2019-12-17}
+
+			$token->save();
+
+			if ( $core_token = $token->get_woocommerce_payment_token() ) {
+				$updated_tokens[] = $core_token->get_token();
+			}
+		}
+
+		return ! empty( $updated_tokens );
```

Comment:
how about we return true/false based on the combined results of `\WC_Data::save()` ? that should return theoretically true/false if save was successful

Reply:
Added c3d078b7d67c7dddd81a7dad21deadf3fe313e35 to tweak the result a bit.

We'll actually probably end up removing this return and phasing out the use of update_tokens as we update the single token handler methods, as it doesn't hold much value either way. There's only one spot where its results are used, via `->add_token()`, so this return mimics that result, i.e. all of the tokens were or were not saved.